src/Nethermind/Nethermind.Facade/Filters/FilterManager.cs

```diff
@@ -251,7 +254,12 @@ private void StoreLogs(LogFilter filter, TxReceipt txReceipt, long logIndex)
 
             if (logFilter.FromBlock.Type == BlockParameterType.Latest || logFilter.ToBlock.Type == BlockParameterType.Latest)
             {
-                //TODO: check if is last mined block
+                // Check if the receipt is from the last mined block
+                Block? latestBlock = _blockFinder.FindLatestBlock();
+                if (latestBlock is null || txReceipt.BlockNumber != latestBlock.Number)
+                {
+                    return null;
+                }
```

Comment:
This is an interesting question if this is correct in engine API driven consensus and how to handle it. Now moving block tree head is decoupled from new payloads and can come only when multiple new payloads pass. Would have to develop some custom pipelining to handle correctly as user would expect each log to be sent. It could  be done with `BlockTree.BlockAddedToMain` event and not current `BlockProcessor.TransactionProcessed`, but probably better is not to change it if no-one complains.